types/leaf_parameters/leaf_parameter_type.go

```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package leaf_parameters
+
+import "fmt"
+
+// LeafParameter represents a parameter type
+// parameter can of concrete (38) or abstract type (P)
+// or another parameterized type like VARCHAR<"L1">
+type LeafParameter interface {
```

Comment:
Why not call this IntegerParameter? Or is this intended to cover type parameters as well? E.g. `STRUCT<T,X>`. 

If it covers type parameters as well, why would it be called leaf? It is only integers, why not use IntegerParameter instead of Leaf?